i-VRESSE/bartended-haddock3
# Use catalog that matches users level

Also loads catalog from disk instead of fetching it from GitHub each time.
Also switched from storing bartender access token in cookie to storing it in a session file with cookie with just session id.

Refs #21 
Refs #15
Fixes #14

This PR requires `main` branch of https://github.com/i-VRESSE/bartender with https://github.com/i-VRESSE/bartender/pull/56 merged in.

Code in app/bartender-client/ is generated with `npm run generate-client` and should not be reviewed on code quality.
The app/catalogs/*,json have created by running `npn run catalogs`.

To test:
1. Configure bartender with roles as described at https://github.com/i-VRESSE/bartended-haddock3/blob/efa9e4da4f35fd086674c4806415571cc0e2d4dc/README.md#haddock3-application
    * Change haddock3 command in config.yaml to cat $config` to mock running of haddock3
1. Spinup 
    * postgresql docker container
    * fill with `alembic upgrade head`
    * bartender serve
    * npm run dev
 3. Regster account
 4. Make account super user with `bartender super <email>`
 5. Logout & login
    *  should see admin page in login menu
6. Goto http://localhost:3000/admin/users and give yourself the guru role.
    * Check on http://localhost:3000/profile that role is assigned
8. At http://localhost:3000/applications/builder check right catalog is loaded.
   * In global parameters and executation section the guru level has `self-contained`, expert has `max_cpus`.
9. Load a worklow (for example https://github.com/i-VRESSE/workflow-builder/blob/main/packages/haddock3_catalog/public/examples/docking-protein-ligand.zip) & submit
10. Job should complete.

When you are not logged in then http://localhost:3000/applications/builder allows download workflow, but not submit (#14).
